### DISCUSSION TITLE
Inconsistent Driver Errors on STM32F0

### DISCUSSION THREAD

**Discussion by JMattChiam** [hathach/tinyusb#2511] [General]
### Operating System

Windows 10

### Board

STM32F072 Custom Board

### Firmware

STM32CubeIDE configured following #633 with tusb_config.h to use RHPORT 0 and USB Full Speed instead of High speed. 

Main firmware: 
examples/device/midi_test
usb_descriptors.c identical to example, other than changing product name. 

### What happened ?

I'm new to USB and trying to get my custom STM32F072 MIDI controller recognized by windows. I've followed the steps outlined in #633 for my needs and the project builds and uploads fine onto the board.
Windows seems to have a hard time with the controller. Sometimes it connects fine and receives the MIDI messages, other times I get driver errors. Infuriatingly, there seems to be a few different behaviors (as shown on USB devices):
- USB Composite device (Working correctly)
- USB Composite device (driver error)
- Actual device name (Working correctly)
- Actual device name (driver error)
- Unknown USB device (Device descriptor request failed)

I've found that for the ones with "driver error", it is encountering error 10 ("This device cannot start"). 
I've attached three wireshark logs if it helps (one for working correctly, one driver error, and one detected as USB composite device and not working). It seems that the one that works correctly includes a SET_INTERFACE request and response, which is missing from both not working ones. 

[Wireshark captures.zip](https://github.com/hathach/tinyusb/files/14494582/Wireshark.captures.zip)
Start Line numbers:
Correct name, driver error: 67
Correct name, working: 63
USB composite device, driver error: 83

Additionally, uninstalling the device on device manager, then clicking "scan for hardware changes" sometimes fixes the issue. 

### How to reproduce ?

NA

### Debug Log as txt file (LOG/CFG_TUSB_DEBUG=2)

NA

### I have checked existing issues, dicussion and documentation

- [X] I confirm I have checked existing issues, dicussion and documentation.

**Comment by HiFiPhile** [comment]
Move to discussion, not TUSB issue.

**Comment by HiFiPhile** [comment]
BTW, you are talking about MIDI but you captured CDC + MSC.

**Reply by JMattChiam** [reply]
What in the captures indicates CDC + MSC? I'm using the stock tusb_config.c from the example with only CFG_TUD_MIDI set as 1 (the rest 0).
I'm mostly confused as to why the behavior is inconsistent.
[Wireshark capture for a Pi pico also running the same midi example, starting from line 63](https://github.com/hathach/tinyusb/files/14504064/Pi.Pico.working.correctly_63.zip)

I've noticed that the capture for the STM32 includes GET_DESCRIPTOR responses with IRP USBD_STATUS: USBD_STATUS_XACT_ERROR (0xc0000011)

**Comment by HiFiPhile** [comment]
> What in the captures indicates CDC + MSC? 

You can see there are MSC and CDC class in `GET DESCRIPTOR Response CONFIGURATION`
![image](https://github.com/hathach/tinyusb/assets/4375114/580357e9-83ae-473f-adff-fe96869564f9)

Also SCSI transfer used by MSC class
![image](https://github.com/hathach/tinyusb/assets/4375114/76b56ea4-3c2a-4a8a-b42d-a6305d09bf15)

>  I'm using the stock tusb_config.c from the example with only CFG_TUD_MIDI set as 1 (the rest 0).

So you did something fundamentally wrong.

**Reply by JMattChiam** [reply]
I'm quite certain that the Mass storage device, along with the SCSI transfers are from the ST-Link debugger I have connected simultaneously. 

Did you look at the lines starting from the connection of my device? 
I am able to see the Audio and MIDI usb descriptors in the wireshark captures that I sent. 
![image](https://github.com/hathach/tinyusb/assets/69444734/5a1527f1-054c-4cf8-a367-f56686e745e6)

[Wireshark with no other USB devices connected](https://github.com/hathach/tinyusb/files/14511029/Wireshark.no.STLink.zip) - The device is plugged in at line 19.

Thank you for helping me through this! I really appreciate it.

**Comment by HiFiPhile** [comment]
> I'm quite certain that the Mass storage device, along with the SCSI transfers are from the ST-Link debugger I have connected simultaneously.

You are right, the capture contains multiple devices.

I suggest you to start with a supported board and official examples, there are too many variables which is impossible to make diagnostic.
(Hardware: power, clock, schematic, layout; Software: cube, custom code)
